lib/jnpr/junos/decorators.py

```diff
@@ -119,14 +119,20 @@ def wrapper(*args, **kwargs):
                             if isinstance(ignore_warn, (str, unicode)):
                                 if not re.search(ignore_warn, err['message'],
```

Comment:
If am bit concerned of using "if not"

say we have error return as
```
<rpc-error>
        <error-severity>warning</error-severity>
        <source-daemon>l2cpd</source-daemon>
        <error-path>[edit protocols]</error-path>
        <error-info>
            <bad-element>lldp</bad-element>
        </error-info>
        <error-message>Advertisement-interval is less than four</error-message>
    </rpc-error>
    <rpc-error>
        <error-severity>warning</error-severity>
        <source-daemon>chassisd</source-daemon>
        <error-path>[edit]</error-path>
        <error-info>
            <bad-element>chassis</bad-element>
        </error-info>
        <error-message>A system reboot is mandatory</error-message>
    </rpc-error>
```
if I give ignore_warning = "A system reboot is mandatory"

during 1st iteration, err['message'] will be "Advertisement-interval is less than four", hence search fails (so "if not" passes) and then throw the exception. 

Are we trying to say that if I am passing ignore_warning as 1 string, then the rpc-error reply from device should also have only one warning, otherwise if there are more than 1 warning are coming from device it should not ignore such response?